packages/wallet-backend/src/payment-pointer/payment-pointer.service.ts

```diff
@@ -23,29 +23,41 @@ export const createPaymentPointer = async (
       req
     )
     const account = await findAccountById(accountId, userId)
-
-    const existingPaymentPointer = await PaymentPointerModel.query().findOne({
+    let paymentPointer = await PaymentPointerModel.query().findOne({
       url: `${env.OPEN_PAYMENTS_HOST}/${paymentPointerName}`
     })
 
-    if (existingPaymentPointer) {
-      throw new ConflictException(
-        'This payment pointer already exists. Please choose another name.'
+    if (paymentPointer) {
+      if (paymentPointer.accountId != accountId || account.userId !== userId) {
+        throw new ConflictException(
+          'This payment pointer already exists. Please choose another name.'
+        )
+      } else if (
+        paymentPointer.accountId === accountId &&
+        account.userId === userId
+      ) {
+        paymentPointer = await PaymentPointerModel.query().patchAndFetchById(
```

Comment:
it can be confusing if you have transactions on the payment pointer (you just created a payment pointer but you have transactions on it)

Reply:
You are right, it can be confusing to create a new payment pointer and already have transactions for it. Let's merge this in and create a separate issue for this and discuss it further.